### ISSUE TITLE
Concurrency is not supported

### ISSUE THREAD

**Issue by fingolfin** [felipec/git-remote-hg#54]
As @felipec stated on issue #16, "Concurrency is not supported".

This should then at least be documented as a known issue, as it can corrupt the internal state of git-remote-hg.

Of course a better solution would be if git-remote-hg used e.g. some kind of locking to prevent multiple processes of accessing the marks files simultaneously. So concurrency would still not be supported, but it would lead to an error, instead of silent (and possibly quite fatal) failure.

**Comment by felipec** [comment]
Ideally it should be Git's core the one that manages marks.

**Comment by fingolfin** [comment]
Yes, it would be nice if one could somehow leverage git's existing lock / concurrency handling.

**Comment by felipec** [comment]
Given the fact that nobody has mentioned this in 9 years I don't think this is a very wanted feature.
